youtube_dl/extractor/drtuber.py

```diff
@@ -44,8 +44,15 @@ def _real_extract(self, url):
         webpage = self._download_webpage(
             'http://www.drtuber.com/video/%s' % video_id, display_id)
 
-        video_url = self._html_search_regex(
-            r'<source src="([^"]+)"', webpage, 'video URL')
+        # the video URL must be extracted from an external JSON stream
+        video_data = self._download_json(
+            'http://www.drtuber.com/player_config_json/?vid=%s&embed=0&aid=0&domain_id=0'
+            % video_id, video_id)
+        # is a high-quality video available?
+        video_url = video_data['files']['hq']
```

Comment:
Correct.  The extractor expects that the ['files', 'hq', 'lq'] keys be present in the JSON data in order to determine the video URL(s).  If this is not the case, something has changed on the server end of things.

How should this scenario be handled?  Should the extractor just raise a new ExtractorError and exit?